### ISSUE TITLE
training stability without pretraining early model

### ISSUE THREAD

**Issue by wyhallenwu** [Xu2729/CoSDH#4]
Hi authors, 

Thank you for your great work and opensource effort!

I'm trying to reproduce your result (though you have provided ckpts). I saw that you pretrain the early model and then finetune it for CoSDH for convergence.

However, I tried without pretraining early model, and find it seems easily to "collapse"  (get NAN in the weights) within 1 or 2 epoches, I want to know if you face any this situations in your trial? Is this common situation so that you have to pretrain with early model?

BTW, this is a great repo with clean code and clear structure! I enjoy reading this code

**Comment by Xu2729** [comment]
After seeing this issue, I immediately retrained the model from scratch (without using any pretrained early fusion model), but I was not able to reproduce the problem. I also did not observe any similar issues in my previous training runs.

Below are the latest TensorBoard logs, where you can see that the loss decreases normally and there is no sign of collapse.

![Image](https://github.com/user-attachments/assets/1418d130-5ee2-4adc-a102-08c6bda7481d)

![Image](https://github.com/user-attachments/assets/c6308ebf-a3ea-4637-be18-ff8eed6b05a2)

![Image](https://github.com/user-attachments/assets/788dfbed-63eb-48ce-9dfa-ea0d38ef46a2)

As a possible (though unlikely) reason, I suspect it might be related to using mixed-precision (FP16) training, which is not enabled by default and requires adding the --half flag manually in the training command. Other than this, I honestly have no further leads regarding this issue. For your reference, here's my training environment:

```txt
Python 3.8.19
torch 1.12.0
numpy 1.24.4
spconv-cu116 2.3.6
cudatoolkit 11.6.2
CUDA Version: 12.4 (nvidia-smi, for NVIDIA GeForce RTX 3080 Ti)
CUDA Version: 12.8 (nvidia-smi, for NVIDIA GeForce RTX 3090)
GPUs: 2*3080Ti or 2*3090
```

Regarding your other question:

>  Is this common situation so that you have to pretrain with early model?

No, this is not a common situation—pretraining with an early-fusion model is not strictly necessary. However, it can help the model consistently converge to better results for certain datasets (such as DAIRV2X). For others, similar results can be achieved even without early-fusion pretraining.

(Thank you for your kind words about the repo—that really made my day!)

**Comment by wyhallenwu** [comment]
Thank you for your quick response and help!

Very interesting! After I move to torch 1.20 and cuda 11.6, everything works fine.

My previous version is cuda11.3 and torch 1.10, maybe this cause the problem.

Again thank you for your effort on maintaining such high quality repo and active reponse! I really appreciate your time for helping!

have a nice day!

**Comment by m-yazgan** [comment]
Hi,
In Opv2v, I can not get good results as you do. In Dair, it was okay, but in OPv2v, the model's performance is poor. Attached is my Logs

<img width="618" height="1082" alt="Image" src="https://github.com/user-attachments/assets/93472a03-a75b-41de-bfdc-5b6b46630b69" />

**Comment by Xu2729** [comment]
> Hi, In Opv2v, I can not get good results as you do. In Dair, it was okay, but in OPv2v, the model's performance is poor. Attached is my Logs
> 
> <img alt="Image" width="618" height="1082" src="https://private-user-images.githubusercontent.com/198627485/549363362-93472a03-a75b-41de-bfdc-5b6b46630b69.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzM1ODg0NjQsIm5iZiI6MTc3MzU4ODE2NCwicGF0aCI6Ii8xOTg2Mjc0ODUvNTQ5MzYzMzYyLTkzNDcyYTAzLWE3NWItNDFkZS1iZmRjLTViNmI0NjYzMGI2OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMzE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDMxNVQxNTIyNDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZjFjZDY1YmU3Zjk2NTJiOWM2ODNlM2IxYjE0NzA1Njc4ZmNlNDY0ZmMzOThkODQ4MzA3YzVkNGE3ZmNjZjAyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ilUNN7xPP_tgiSgd-fKB1CGCF7orHQpT-mTw9ES5ZKk">

Based on the training logs you provided, the loss curves look normal. The number of steps appears different (likely due to different batch sizes or number of GPUs), but that should not significantly affect the results. Given the current information, it’s difficult for us to pinpoint the exact cause — it might, as before, be related to versions of critical dependencies. We recommend ensuring your dependency versions match those we provided in our earlier reply. Also, if you can share the versions of the dependencies you used and the inference results, that would help us investigate the issue.
